src/test/java/com/google/sps/servlets/UserDataServletTest.java

```diff
@@ -0,0 +1,256 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.gson.Gson;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+/** Tests the User Data Servlet */
+@RunWith(JUnit4.class)
+public final class UserDataServletTest {
+  private static final String USER_BIO_PROPERTY = "bio";
+  private static final String USER_EMAIL_PROPERTY = "email";
+  private static final String USER_FOUND_PROPERTY = "user-found";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+
+  // Test User info
+  private static final String TEST_USER_NAME = "Tim";
+  private static final String TEST_USER_ID = "123";
+  private static final String TEST_USER_EMAIL = "tim@gmail.com";
+  private static final String TEST_USER_BIO = "Amazing!";
+  private static final String[] TEST_USER_FRIENDS_LIST = new String[]{"321"};
+  private static final String ALTERNATE_TEST_USER_NAME = "John";
+
+  private final Gson gson = new Gson();
+  // Uses a local datastore stored in memory for tests
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock private HttpServletRequest mockRequest;
+  @Mock private HttpServletResponse mockResponse;
+  private UserDataServlet servletUnderTest;
+  private StringWriter stringWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new UserDataServlet();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the doGet method, where the datastore doesn't contain the test user id.
+   * Expected output: JSON response with "user-found" property as false.
+   */
+  @Test
+  public void testGetMethodWithInvalidId() throws Exception {
+    // Mock the getParameter call
+    when(mockRequest.getParameter(USER_ID_PROPERTY)).thenReturn(TEST_USER_ID);
+
+    // Store output given by mockResponse
+    stringWriter = new StringWriter();
+    writer = new PrintWriter(stringWriter);
+    when(mockResponse.getWriter()).thenReturn(writer);
+    when(mockResponse.getContentType()).thenReturn("application/json");
```

Comment:
There might be a good reason here, but why do we need to mock the content type? You set it yourself in the doGet(). If you need to check that the content type is correct, you can't do that with this (and in reality you don't need that exact check).

Reply:
Makes sense that it's not necessary. Removed.